modules/drivers/athena/src/metabase/driver/athena.clj

```diff
@@ -255,6 +257,18 @@
   [_driver _semantic-type expr]
   (hx/->time expr))
 
+(defmethod sql.qp/->honeysql [:athena :datetime-diff]
+  [driver [_ x y unit]]
+  (let [x (sql.qp/->honeysql driver x)
```

Comment:
This is missing the error checking that all of the other drivers have

Reply:
Unless athena doesn't have any type restrictions? If that's the case, it's probably worth calling out in a comment, because it seems like all the other driver implementations do have restrictions, I've come to expect them when reading implementations of this method

Reply:
Thanks for catching this. I left out the error checking because I believed Athena didn't support the time type because `supports-time-type?` is false for athena. I missed the fact that it actually can have a `time` type from literals. Time literals fed to datetimeDiff directly as strings will fail at the MBQL schema checking stage, but with nested questions a time literal can still sneak through as a field.

I just realised athena's datetimeDiff is not being tested for the `timestamp with time zone` type either because `supports-timestamptz-type?` also returns false. I'll fix this and put this PR back up for review.